src/nitpackage.nit

```diff
@@ -287,6 +393,109 @@ redef class ConfigTree
 	end
 end
 
+# A Makefile for the Nit project
+class NitMakefile
+
+	# Nit files to compile
+	var nit_files: Array[String]
+
+	# List of rules to add in the Makefile
+	fun rules: Array[MakeRule] do
+		var rules = new Array[MakeRule]
+
+		var rule_all = new MakeRule("all")
+		if nit_files.not_empty then
+			rules.add rule_all
+		end
+
+		for file in nit_files do
+			var bin = file.basename.strip_extension
+
+			rule_all.deps.add bin
+
+			var rule = new MakeRule(bin)
+			rule.lines.add "mkdir -p bin/"
+			rule.lines.add "$(NITC) {file} -o bin/{bin}"
+			rules.add rule
```

Comment:
Could-you add a dependency on sources files. so we can have

~~~
nitmd: $(shell $(NITLS) -M nitmd.nit)
	mkdir -p bin/
	$(NITC) nitmd.nit -o bin/nitmd
~~~

instead of

~~~
nitmd:
	mkdir -p bin/
	$(NITC) nitmd.nit -o bin/nitmd
~~~